client/components/category-navigation/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { Button } from '@wordpress/components';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import classnames from 'classnames';
+
+import './style.scss';
+
+type CategoryNavProps = {
+	buttons?: {
+		icon: string;
+		label: string;
+		link: string;
+	}[];
+	children?: React.ReactNode;
+	list: {
+		name: string;
+		label?: string;
+		link: string;
+	}[];
```

Comment:
Let's just call `usePatternCategories` inside this component instead.

Reply:
It's not final version, I will adjust it to be used as a common component for different places.